### PR TITLE
Ridiculous-sized PR about windows.

### LINKED ISSUE #490: you can leap over barricades into windows

### LINKED ISSUE #568: hivelords can build walls inside windows
(but not window frames)

### PR DESCRIPTION
**Pull request by Ghommie**
- most probably fixes #490 and fixes #568, didn't even care testing on it though.
- Sheets of glass are less snowflakey and also make good use of stack_recipes lists now.
- Diverge paths for reinforced and phoron glass types to avoid problems.
- `/obj/structure/window` is just the parent type for both framed and frameless windows now: `/obj/structure/window/base` is now the path used for frameless windows, to avoid icon and smoothing conflicts with framed ones.
- Icon smoothing for frameless windows has been brought back. The old devs seemengly removed it when switching to framed windows.
- Spawner subpaths for directional, map spawned window. Use them, mappers.
- Repathed frameless windows on not special or archived maps. That's the source of the thousands lines. Tests appreciated.
- Changes to window (dis)assembling steps to be similar to /TG/station. For framed windows they are:
- screwdriver <-> crowbar <-> wrench, instead of simply screwdriving.
- Other minor things.
- Updated `_HELPERS/lists.dm`, removed the old `_math.dm` from _DEFINES, set the milestone for `colors.dm` and `construction.dm`.

### CODE REVIEW

File: code/game/objects/structures/door_assembly.dm [hunk 1]
```diff
@@ -163,7 +163,7 @@ obj/structure/door_assembly
 				if(do_after(user, 40, TRUE, 5, BUSY_ICON_BUILD))
 					if(!src || !WT.isOn()) return
 					to_chat(user, "\blue You welded the glass panel out!")
-					new /obj/item/stack/sheet/glass/reinforced(src.loc)
+					new /obj/item/stack/sheet/rglass(src.loc)
```
Comment (Rohesie) [inline code/game/objects/structures/door_assembly.dm]:
```suggestion
					new /obj/item/stack/sheet/rglass(loc)
```

**Review by Rohesie — CHANGES_REQUESTED** [review]

**Comment by Rohesie** [comment]
This conflicts with an Ice rework and will conflict with #569 's Almayer minor change. After that we can test and eventually merge this, though. Please test your changes in the meantime.

File: ColonialMarinesALPHA.dme [hunk 2]
```diff
@@ -15,11 +15,12 @@
 #include "code\stylesheet.dm"
 #include "code\world.dm"
 #include "code\__DEFINES\__game.dm"
-#include "code\__DEFINES\_math.dm"
```
Comment (Rohesie) [inline ColonialMarinesALPHA.dme]: Wha?

**Review by Rohesie — CHANGES_REQUESTED** [review]

File: ColonialMarinesALPHA.dme (hunk 2 above)
Comment (Ghommie) [reply ColonialMarinesALPHA.dme]: It's an old file, when I defined maths.dm, I cut out the many mathematical operations here into the new one, only leaving those bits that are actually off topic.

**Pull request closed without merging**